### PR TITLE
fix string concat for Nix

### PR DESCRIPTION
**Pull request by francois-caddet**
fix string concat in Nix.

### CODE REVIEW

File: stdlib/compat.ncl [hunk 1]
```diff
@@ -1,6 +1,14 @@
 {
 compat | doc "Nix compatibility layer. should not be used by Nickel program."
```
Comment (yannham) [inline stdlib/compat.ncl:2]:
```suggestion
compat | doc "Nix compatibility layer. This library should not be used by Nickel program."
```

File: stdlib/compat.ncl [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
 = {
+
+  # the adition in Nix can be a string concataination.
```
Comment (yannham) [inline stdlib/compat.ncl:5]:
```suggestion
  # The addition in Nix can be a string concatenation.
```

File: stdlib/compat.ncl [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+  # this function handle this behaviour.
```
Comment (yannham) [inline stdlib/compat.ncl:6]: This is not a very informative comment. Maybe something in the line of:

> Addition in Nix is overloaded to work both as number addition and string concatenation.
> There is no such operator in Nickel. This function implement the equivalent of the Nix primitive operator by dynamically
> dispatching between addition and concatenation, based on the runtime type of its arguments.

**Review by yannham — COMMENTED** [review]
Nitpick: the `compat` module of stdlib should maybe mention Nix in its name. It's not obvious what "compatibility" it refers to. Maybe `nix_compat` ? Same for the stdlib module, I think.

**Comment by francois-caddet** [comment]
> 

Yes, it'll better with something like `__nix_compat__".
The issue is that this "compat" lib is already in `nix/nickel` branch. So may we fix it in another PR?

**Comment by yannham** [comment]
> So may we fix it in another PR?

Sounds good.

**Review by yannham — APPROVED** [review]

**Pull request merged**